internal/backend/sftp/sftp.go

```diff
@@ -354,14 +354,13 @@ func (r *SFTP) Save(ctx context.Context, h restic.Handle, rd restic.RewindReader
 			debug.Log("sftp: failed to remove broken file %v: %v",
 				f.Name(), rmErr)
 		}
-
-		err = r.checkNoSpace(dirname, rd.Length(), err)
```

Comment:
Overwriting `err` probably has no effect as it isn't a named return parameter. That is, the code currently only overwrites the local variable `err` but has no effect on the returned value.